src/System.Text.Json/src/System/Text/Json/JsonReaderState.cs

```diff
@@ -62,12 +62,12 @@ public struct JsonReaderState
         /// across async/await boundaries and hence this type is required to provide support for reading
         /// in more data asynchronously before continuing with a new instance of the <see cref="Utf8JsonReader"/>.
         /// </remarks>
-        public JsonReaderState(int maxDepth = StackFreeMaxDepth, JsonReaderOptions options = default)
+        public JsonReaderState(int maxDepth = AllocationFreeMaxDepth, JsonReaderOptions options = default)
         {
             if (maxDepth <= 0)
                 throw ThrowHelper.GetArgumentException_MaxDepthMustBePositive();
 
-            _stackFreeContainer = default;
+            _allocationFreeContainer = default;
```

Comment:
It would be cleaned to move this into your CustomUncheckedBitArray now that you have one.

Reply:
That's a good idea but that requires a larger change so I will do so separately (I don't want to unblock this PR on it).
https://github.com/dotnet/corefx/issues/33518

Reply:
Well, we typically try to make the code right in the first PR. I do not think it would require that much extra code compared to what you have in this PR.

Reply:
Right now, you got what is essentially a stack of bits with the state spread over `_allocationFreeContainer`, `_currentDepth`, `CustomUncheckedBitArray _bitArray`. Either drop the `CustomUncheckedBitArray` completely a just inline everything into the `JsonReaderState`; or make a proper type for a stack of bits.

Reply:
OK. I moved `_allocationFreeContainer` into `CustomUncheckedBitArray`